### PR TITLE
[CR] Change player spriteflipping trigger from hardcoded keys to plmove

### LINKED ISSUE #27926: Character's Sprite flips around when using Arrowkeys but not Numpad
**Describe the bug**  
Recently, a update made your character's and monster's sprites face left or right depending on where they moved from. This happens to your character when you move with the arrow keys, but not with the numpad for some reason.

**To Reproduce**  
Try moving with both arrow keys and numpad and see the difference

**Expected behavior**  
I kinda expected the numpad to flip my sprite around, but it doesn't. It really should. 

**Versions and configuration(please complete the following information):**
 - OS: Windows 10
 - Game Version: 0.C-37114-g7e78b20
 - Graphics version: Tiles
 - Mods loaded: Default Mods

This is a pretty sweet feature and I really didn't know this existed for a bit since I use the numpad for moving so much.

### PR DESCRIPTION
**Pull request by ThinkInvis**
#### Summary
SUMMARY: Bugfixes "Make player spriteflip happen on player move, instead of directly on left/right keypress"

#### Purpose of change
Fixes #27926 -- Character's Sprite flips around when using Arrowkeys but not Numpad.

#### Describe the solution
Player spriteflipping was previously hardcoded to the left and right arrow keys during `player.process_turn`. The solution: instead triggering spriteflip on any attempted Δx during `game.plmove`. This should cause sprite flipping whenever the player attempts to move, or stumbles about while stunned, through _any_ means.
[CR] Also, this will stop the player from spriteflipping while turning a vehicle left/right... which probably shouldn't be happening in the first place, as there's a separate indicator for that, but I don't know for sure. There's a few other potential problems with the order of operations here: should some plmove effects be left out of spriteflip consideration (e.g. being stunned)? Should others be included (e.g. being unable to move away during effect_amigara)?

#### Describe alternatives you've considered
The original issue could also be resolved by hardcoding more keys (numpad 741 for left, numpad 963 for right) to cause player spriteflip.

### CODE REVIEW

**Comment by ThinkInvis** [comment]
Could've sworn I fixed those parentheses during the first commit...

File: src/game.cpp [hunk 1]
```diff
@@ -10179,6 +10179,14 @@ bool game::plmove( int dx, int dy, int dz )
         }
     }
 
+    // If the player is *attempting to* move on the X axis, update facing direction of their sprite to match.
+    int new_dx = dest_loc.x - u.posx();
```
Comment (Night-Pryanik) [inline src/game.cpp]: Could be const.

**Comment by ThinkInvis** [comment]
Just realized that the primary change made by this PR removes a few more spriteflip-triggering actions than just steering vehicles -- for example, opening doors, examining things.... That's another point for the hardcoded-keys approach, but I feel like the move-effect approach is still better (as the former makes it much harder to stop the hardcoded keys from causing spriteflip, if necessary). It looks like, as shown in the most recent commit, action.cpp's `choose_direction` method is a centralized resource for such actions. The fire(/throw/reach-attack) action is a separate thing, but I think I also managed to find a good place to add a spriteflip handler there. 

New [CR] things: I did some testing and couldn't find any problems with adding spriteflip handlers in either of these places, but that doesn't mean those problems don't exist. Also, the current direction I'm taking with this is that the sprite turns to face the player's _consideration_ of a new action, e.g. towards target while firing; meanwhile, the original feature only triggered once a turn passed. Latter is more realistic, former can be more aesthetically pleasing.

**Comment by ifreund** [comment]
This looks great man, the lack of sprite flipping on diagonal movement had almost annoyed me to the point that I opened my own pr.  Dehardcoding these is definitely a lot cleaner.

>Also, the current direction I'm taking with this is that the sprite turns to face the player's consideration of a new action, e.g. towards target while firing; meanwhile, the original feature only triggered once a turn passed. Latter is more realistic, former can be more aesthetically pleasing.

I think this is the right approach, and I would say the point of having sprites at all is to be aesthetically pleasing.

**Comment by rumly111** [comment]
so, this fix won't appear until 0.D is released ?

**Comment by kevingranade** [comment]
Spriteflipping isn't in 0.D, if it was I'd merge this as well.

**Pull request merged**